src/index.js

```diff
@@ -31,26 +31,100 @@ function loadAppId(callback) {
     });
 }
 
+function getUrl() {
+    var server_url = localStorage.getItem('config.server_url') || 'frontend.binaryws.com';
+    return 'wss://' + server_url + '/websockets/v3';
+}
+
+function processRedirect(selected_language_name) {
+    loadAppId(function(err, app_id_json) {
+        var ip_country;
+        var account_list;
+        var app_id = err ? default_app_id : getAppId(app_id_json);
+        var api_url = getUrl() + '?l=' + selected_language_name + '&app_id=' + app_id;
+        var ws = new WebSocket(api_url);
+        ws.onopen = sendWebsiteStatus;
+        ws.onmessage = processApiResponse;
+
+        function getAppId(app_id_json) {
+            var stored_app_id = '';
+
+            for(var url in app_id_json) {
+                if(href.lastIndexOf(url, 0) === 0) {
+                    stored_app_id = app_id_json[url];
+                }
+            }
+
+            return stored_app_id || default_app_id;
+        }
+        getUrl();
+
+        function sendApiRequest(request) {
+            ws.send(JSON.stringify(request));
+        }
+
+        function sendWebsiteStatus(evt) {
+            sendApiRequest({ website_status: 1 });
+        }
+
+        function sendAuthorize(token) {
+            sendApiRequest({ authorize: token });
+        }
+        
+        function processApiResponse(response) {
+            var data = JSON.parse(response.data);
+
+            if(data.website_status){
+                ip_country = data.website_status.clients_country;
+                if (local_storage.get('oauth')) {
+                    var token = local_storage.get('oauth')[0].token;
+                    sendAuthorize(token);
+                } else {
+                    if(isEuCountrySelected(ip_country)) {
+                        window.location.href = getBinaryUrlWithoutLng('move-to-deriv');
```

Comment:
```suggestion
                        window.location.href = getBinaryUrlWithoutLng('move-to-deriv');
```
This can be extracted to its own function to make it clearer